api/base/serializers.py

```diff
@@ -1058,9 +1058,24 @@ def is_valid(self, clean_html=True, **kwargs):
         return ret
 
 
-class PrefetchRelationshipsSerializer(ser.Serializer):
+class SparseFieldsetMixin(object):
+    def parse_sparse_fields(self, allow_unsafe=False, **kwargs):
+        request = kwargs.get('context', {}).get('request', None)
+        if request and (allow_unsafe or request.method in permissions.SAFE_METHODS):
+            sparse_fieldset_query_param = 'fields[{}]'.format(self.Meta.type_)
+            if sparse_fieldset_query_param in request.query_params:
+                fieldset = request.query_params[sparse_fieldset_query_param].split(',')
+                for field_name in self.fields.fields.copy().keys():
+                    if field_name in ('id', 'links', 'type'):
+                        # MUST return these fields
+                        continue
+                    if field_name not in fieldset:
+                        self.fields.pop(field_name)
+
+class PrefetchRelationshipsSerializer(ser.Serializer, SparseFieldsetMixin):
```

Comment:
I agree that it doesn't make much conceptual sense, but it was that or change the inheritance and initialization for the other things that inherit from `PrefetchRelationshipsSerializer` that aren't `JSONAPISerializer`:

```
api/base/serializers.py			|1069 col 1|	class JSONAPISerializer(PrefetchRelationshipsSerializer):
api/base/serializers.py			|1251 col 1|	class JSONAPIRelationshipSerializer(PrefetchRelationshipsSerializer):
api/base/serializers.py			|1354 col 1|	class LinkedNodesRelationshipSerializer(PrefetchRelationshipsSerializer):
api/base/serializers.py			|1419 col 1|	class LinkedRegistrationsRelationshipSerializer(PrefetchRelationshipsSerializer):
api/institutions/serializers.py		|60 col 1|	class InstitutionNodesRelationshipSerializer(PrefetchRelationshipsSerializer):
api/nodes/serializers.py		|49 col 1|	class NodeLicenseSerializer(PrefetchRelationshipsSerializer):
api/nodes/serializers.py		|1002 col 1|	class NodeInstitutionsRelationshipSerializer(PrefetchRelationshipsSerializer):
api/users/serializers.py		|168 col 1|	class UserInstitutionsRelationshipSerializer(PrefetchRelationshipsSerializer):
api/view_only_links/serializers.py	|54 col 1|	class ViewOnlyLinkNodesSerializer(PrefetchRelationshipsSerializer):
```
It might make more sense if `PrefetchRelationshipsSerializer` was called something like `BaseAPISerializer`, as that appears to be its role.